### PR TITLE
add: support banner themes in the `vexide::main` macro

### PR DESCRIPTION
**Pull request by Gavin-Niederman**
## Describe the changes this PR makes. Why should it be merged?
Adds many new banner themes and the ability to customize them in the vexide main macro. This PR contains breaking changes to the main macro due to the new syntax.

Synthwave banner theme:
image

Banner disabling syntax:
image

Banner theme syntax:
image

## Additional Context
- These changes update the crate's interface (e.g. functions/modules added or changed).
- I have tested these changes on a VEX V5 brain.
- These are breaking changes (semver: major).

### CODE REVIEW

**Review by doinkythederp — CHANGES_REQUESTED** [review]
Wrong button

File: packages/vexide-startup/src/banner/themes.rs [hunk 1]
```diff
@@ -1,3 +1,46 @@
...
+//!        "\u{1b}[1;38;2;136;192;208m",
+//!        "\u{1b}[1;38;2;136;192;208m",
+//!        "\u{1b}[1;38;2;136;192;208m",
+//!    ],
+//!    logo_secondary: "\x1B[38;5;254m",
+//!    crate_version: "[1;33m",
+//!    metadata_key: "[1;33m",
+//!};
+//!
+//! #[vexide::main(banner(enabled = true, theme = CUSTOM_THEME))]
```
[the remark this answers is not in the document]
Comment (Gavin-Niederman) [reply packages/vexide-startup/src/banner/themes.rs:40]: I consider that, but most people will just be using (enabled = false) or (theme = theme) and doing that with an option seems less readable to me

File: packages/vexide-startup/src/banner/themes.rs [hunk 2]
```diff
@@ -1,3 +1,46 @@
...
+//! - [Aro Ace flag](THEME_ARO_ACE)
+//! - [Nonbinary flag](THEME_NONBINARY)
+//! - [Bisexual flag](THEME_BISEXUAL)
+//! - [Trans flag](THEME_TRANS)
+//! - [Tidal wave colors](THEME_TIDAL_WAVE)
+//! - [Whitescreen theme](THEME_WHITESCREEN)
+//! - [America flavored theme](THEME_MURICA)
+//!
+//! Custom themes can be created by defining a constant of type [`BannerTheme`] with the desired ANSI escapes.
+//! This theme can then be passed to the [`vexide::main`](https://docs.rs/vexide/latest/vexide/attr.main.html).
```
[the remark this answers is not in the document]
Comment (Gavin-Niederman) [reply packages/vexide-startup/src/banner/themes.rs:18]: Yeah I meant to say vexide::main macro

**Comment by doinkythederp** [comment]
Also BTW this makes the docs out of date for the `vexide::main` macro

File: packages/vexide-macro/src/lib.rs [hunk 3]
```diff
@@ -147,6 +148,17 @@ fn make_entrypoint(opts: MacroOpts) -> proc_macro2::TokenStream {
 /// }
 /// ```
 ///
+/// ```ignore
+/// # #![no_std]
+/// # #![no_main]
+/// # use vexide::prelude::*;
+/// use vexide::startup::banner_themes::THEME_SYNTHWAVE;
+/// #[vexide::main(theme = THEME_SYNTHWAVE)]
```
Comment (doinkythederp) [inline packages/vexide-macro/src/lib.rs:156]: Should this be `main(banner(theme = THEME_SYNTHWAVE))`? Also see line 145.

**Review by doinkythederp — CHANGES_REQUESTED** [review]

File: packages/vexide-macro/src/lib.rs (hunk 3 above)
Comment (Gavin-Niederman) [reply packages/vexide-macro/src/lib.rs]: fixed

File: packages/vexide-startup/src/banner/mod.rs [hunk 4]
```diff
@@ -1,3 +1,10 @@
+//! vexide startup banner.
+//!
+//! The banner can be printed with a theme using the [`print`] function.
```
Comment (doinkythederp) [inline packages/vexide-startup/src/banner/mod.rs:3]: IMO this should mention that the banner is automatically printed when using the `vexide::main` macro

Comment (Gavin-Niederman) [reply packages/vexide-startup/src/banner/mod.rs]: done

**Review by doinkythederp — APPROVED** [review]

**Pull request merged**